app/helpers/repository_helper.rb

```diff
@@ -109,7 +109,7 @@ def remove_files(files, user, repo, path: '/', txn: nil)
       basename = sanitize_file_name(basename)
       file_path = current_path.join(subdir_path).join(basename)
       file_path = file_path.to_s
-      txn.remove(file_path, current_revision)
+      txn.remove(file_path, current_revision, keep_folder)
```

Comment:
Please change the `remove` call here to also pass the keep_folder value as a keyword argument:

```
txn.remove(file_path, current_revision, keep_folder: keep_folder)
```